mobile-app/app/components/DfxDexFeeInfo.tsx

```diff
@@ -0,0 +1,88 @@
+import { useFeatureFlagContext } from '@contexts/FeatureFlagContext'
+import { OwnedTokenState, TokenState } from '@screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreen'
+import { useDexStabilization } from '@screens/AppNavigator/screens/Dex/hook/DexStabilization'
+import { WalletToken } from '@store/wallet'
+import { translate } from '@translations'
+import { useEffect, useMemo } from 'react'
+import { Linking, StyleProp, ViewStyle } from 'react-native'
+import { TouchableOpacity } from 'react-native-gesture-handler'
+import { InfoText } from './InfoText'
+
+interface DfxDexFeeInfoProps {
+  token: WalletToken
+  style?: StyleProp<ViewStyle>
+  getDexFee?: (fee: string) => void
+}
+
+export function DfxDexFeeInfo (props: DfxDexFeeInfoProps): JSX.Element {
+  const dfi: TokenState = useMemo(() => {
+    return {
+      id: '0_unified',
+      reserve: '',
+      displaySymbol: 'DFI',
+      symbol: 'DFI'
+    }
+  }, [])
+
+  // const token: OwnedTokenState = useMemo(() => {
+  //   return {
+  //     id: props.token.id,
+  //     reserve: '',
+  //     displaySymbol: props.token.displaySymbol,
+  //     symbol: props.token.symbol,
+  //     amount: props.token.amount
+  //   }
+  // }, [props.token])
+
+  const dusd: OwnedTokenState = useMemo(() => {
+    return {
+      id: '11',
+      reserve: '',
+      displaySymbol: 'DUSD',
+      symbol: 'DUSD',
+      amount: 'props.token.amount'
+    }
+  }, [])
+
+  // dex stabilization
+  const { isFeatureAvailable } = useFeatureFlagContext()
+  const isDexStabilizationEnabled = isFeatureAvailable('dusd_dex_high_fee')
+  const {
+    dexStabilization: {
+      // dexStabilizationType,
+      dexStabilizationFee
+    }
+  } = useDexStabilization(dusd, dfi)
+
+  const isCrypto = (): boolean => ['DFI', 'dBTC', 'dETH', 'dUSDT', 'dUSDC', 'dLTC', 'dBCH', 'dDOGE'].includes(props.token.displaySymbol)
+
+  const returnDexFee = (): void => {
+    if (isCrypto()) {
+      props.getDexFee?.('0')
+    } else {
+      props.getDexFee?.(dexStabilizationFee)
+    }
+  }
+
+  returnDexFee()
+
+  useEffect(() => {
+    returnDexFee()
+  }, [props.token])
```

Comment:
Is this really a good approach? 😉  Wouldn't it be easier to have a separate reusable hook for this?